swingset/src/main/java/com/nqadmin/swingset/SSDBComboBox.java

```diff
@@ -1343,7 +1313,11 @@ public void setSelectedItem(final Object _value) {
 		//
 		// 2020-10-03_BP: Updated getSelectedValue() to properly return the primary key rather than using getSelectedIndex() during a call to this method.
 		// Only try to update item for a valid list item.
-			if (selectedItem!=null) {
+
+			// TODO: when this method is called internally,
+			//       newSelectedItem may be nullItem. Should
+			//       the other branch be taken in that case?
+			if (newSelectedItem!=null) {
```

Comment:
?

Reply:
If the user has selected nullItem then I think the code should proceed as though there has been a valid selection, not null. I think null was expected in the case where the user typed gibberish into a GlazedList combo.

Reply:
Notice the `==null` branch does `setSelectedItem(nullItem)` so nullItem is selected in both branches for either null or nullItem. Only the log entry is different and there's enough log info to know. So just getting rid of the cast should be ok.